app/src/components/Sidebar.tsx

```diff
@@ -21,35 +21,62 @@ class Sidebar extends EnhancedComponent<ISidebarProps, ISidebarState> {
         {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
     ]
 
+    private shownGenres: ISidebarGenreChannel[] = [];
+
     protected constructor(props: ISidebarProps) {
         super(props);
         this.state = {
-            searchValue: "",
             genre: "",
             icon: "",
+            shownGenres: [
+                {genre: "Electronic", icon: "https://img.icons8.com/ios-glyphs/30/000000/electronic-music.png"},
+                {genre: "Rock", icon: "https://img.icons8.com/ios-glyphs/30/000000/rock-music.png"},
+                {genre: "Lo-Fi", icon: "https://img.icons8.com/ios-glyphs/30/000000/easy-listening.png"},
+                {genre: "Reggae", icon: "https://img.icons8.com/ios-glyphs/30/000000/reggae.png"},
+                {genre: "Country", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+                {genre: "Hip-Hop", icon: "https://img.icons8.com/ios-glyphs/30/000000/hip-hop-music.png"},
+                {genre: "Jazz", icon: "https://img.icons8.com/ios-glyphs/30/000000/saxophone.png"},
+                {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
+            ],
         };
+        this.onSearch = this.onSearch.bind(this);
     }
 
     private onSearch(event: React.SyntheticEvent) {
+        event.preventDefault();
         let target = event.currentTarget as HTMLInputElement;
-        let value = target.value;
-        this.setState({
-            searchValue: value
-        });
+        let searchValue = target.getElementsByClassName("search_input")[0].getAttribute("value");
+        if (searchValue !== "") {
+            for (let i: number = 0; i < this.musicGenres.length; i++) {
+                if (this.musicGenres[i].genre.includes(searchValue)) {
+                    this.shownGenres.push(this.musicGenres[i]);
```

Comment:
This is failing. Try this to reproduce the error:
1. type in "Rock", hit enter to search, you will see Rock filtered out
2. delete everything from the text box, hit enter, you will see all genres
3. type in "Rock", hit enter to search again, now you will find two instances of Rock
4. continue doing this and this keeps multiplying:
<img width="310" alt="Screen Shot 2020-06-16 at 6 12 10 PM" src="https://user-images.githubusercontent.com/7976828/84844089-81a74b00-affe-11ea-9db8-ac70779aef29.png">